### PR TITLE
Add data checks to user input

### PR DESCRIPTION
**Pull request by cbosdo**
## What does this PR change?

Add length and content check to user input. Constraints extracted from https://github.com/uyuni-project/uyuni/blob/master/java/code/src/com/redhat/rhn/frontend/action/multiorg/validation/orgCreateForm.xsd#L25

## Test coverage
- Unit tests were added

- [X] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/23337

- [X] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [X] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: mgradm/cmd/install/shared/flags.go [hunk 1]
```diff
@@ -56,30 +60,50 @@ type InstallFlags struct {
...
 }
 
+// idChecker verifies that the value is a valid identifier.
+func idChecker(value string) bool {
+	isAscii := regexp.MustCompile(`^[[:graph:]]+$`)
+	if isAscii.MatchString(value) && !strings.ContainsAny(value, "+&'\"`=#") {
+		return true
+	}
+	fmt.Println("Can only contain ASCII characters without +&'\"`=#")
+	return false
```
Comment (aaannz) [inline mgradm/cmd/install/shared/flags.go:70]: So do we allow e.g. `>` in the id?

Wouldn't `[[:alnum:]]` be better here?

File: mgradm/cmd/install/shared/flags.go [hunk 2]
```diff
@@ -56,30 +60,50 @@ type InstallFlags struct {
...
+}
+
+// emailChecker verifies that the value is a valid email address.
+func emailChecker(value string) bool {
+	r := regexp.MustCompile(`^(.+)@(.+)(\..+)*$`)
+	if !r.MatchString(value) {
+		fmt.Println("Not a valid email address")
+		return false
+	}
+	return true
```
Comment (aaannz) [inline mgradm/cmd/install/shared/flags.go:80]: This is quite naive, I understand it's taken from Java part, but we have https://pkg.go.dev/net/mail#ParseAddress

File: shared/utils/utils.go [hunk 3]
```diff
@@ -19,15 +19,41 @@ import (
...
+		tmpValue := strings.TrimSpace(string(bytePassword))
+		r := regexp.MustCompile(`^[^\t ]+$`)
+		validChars := r.MatchString(tmpValue)
+		if !validChars {
+			fmt.Printf("Cannot contain spaces or tabs")
+		}
+
+		if validChars && checkValueSize(tmpValue, min, max) {
+			*value = tmpValue
+		}
```
Comment (aaannz) [inline shared/utils/utils.go:56]: I am sorry, but this is some go-fu which I am not familiar with.
What does `for *value == "" {` should do? How can this function fail if the format is not valid?

File: mgradm/cmd/install/shared/flags.go (hunk 1 above)
Comment (cbosdo) [reply mgradm/cmd/install/shared/flags.go:70]: well, in the Java code it's even worse, we also include control characters 😉 But yes, I could just change to `[[:alnum:]]`

File: shared/utils/utils.go (hunk 3 above)
Comment (cbosdo) [reply shared/utils/utils.go:56]: This is like `for ; *value == "" ; {`. This only stops with a correct value or a `Ctrl-C`.

**Pull request merged**